pallets/market/src/lib.rs

```diff
@@ -1002,15 +834,15 @@ pub mod pallet {
         ) -> Result<Cid, DispatchError> {
             let pieces = proposals
                 .map(|p| {
-                    let cid = p.cid()?;
-                    let commitment =
-                        Commitment::from_cid(&cid).map_err(|err| ProposalError::CommD(err))?;
+                    let cid = p.cid().map_err(|_| PaddedPieceSizeError::InvalidPieceCid)?;
+                    let commitment = Commitment::from_cid(&cid)
+                        .map_err(|_| PaddedPieceSizeError::UnableToCreateCommD)?;
```

Comment:
Same as the previous one, is it because of Display?

Reply:
We are collecting errors from the commitment here, any error that happens will be logged and returned as an invalid commD error. I would add the error to the error enum in but I need `PalletError` implemented for `PaddedPieceSizeError`. I think logging invalid commd + docs should be enough for the user.